### PR TITLE
core: fix dependencies, update nix

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,8 +1,11 @@
 {
   description = "nix flake for simplex-chat";
   inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable"; #angerman/nixpkgs/patch-1"; # based on 21.11, still need this, until everything is merged into 21.11.
-  inputs.haskellNix.url = "github:input-output-hk/haskell.nix?ref=angerman/try-no-libcharset";
-  inputs.haskellNix.inputs.nixpkgs.follows = "nixpkgs";
+  inputs.hackage = {
+    url = "github:input-output-hk/haskell.nix";
```
Comment (angerman) [inline flake.nix:5]: This should point to `hackage.nix`

File: flake.nix [hunk 2]
```diff
@@ -1,8 +1,11 @@
 {
   description = "nix flake for simplex-chat";
   inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable"; #angerman/nixpkgs/patch-1"; # based on 21.11, still need this, until everything is merged into 21.11.
-  inputs.haskellNix.url = "github:input-output-hk/haskell.nix?ref=angerman/try-no-libcharset";
-  inputs.haskellNix.inputs.nixpkgs.follows = "nixpkgs";
```
Comment (angerman) [inline flake.nix:5]: We want to keep these.

**Review by angerman — CHANGES_REQUESTED** [review]
We also want a line:

inputs.haskellNix.inputs.hackage = "hackage";

To ensure that haskell.nix knows about the new hackage.nix

File: flake.nix (hunk 2 above)
Comment (epoberezkin) [reply flake.nix:5]: both lines?

**Pull request merged**